types/context.go

```diff
@@ -41,29 +41,31 @@ type Context struct {
-func (c Context) MultiStore() storetypes.MultiStore          { return c.ms }
-func (c Context) BlockHeight() int64                         { return c.header.Height }
-func (c Context) BlockTime() time.Time                       { return c.header.Time }
-func (c Context) ChainID() string                            { return c.chainID }
-func (c Context) TxBytes() []byte                            { return c.txBytes }
-func (c Context) Logger() log.Logger                         { return c.logger }
-func (c Context) VoteInfos() []abci.VoteInfo                 { return c.voteInfo }
-func (c Context) GasMeter() storetypes.GasMeter              { return c.gasMeter }
-func (c Context) BlockGasMeter() storetypes.GasMeter         { return c.blockGasMeter }
-func (c Context) IsCheckTx() bool                            { return c.checkTx }
-func (c Context) IsReCheckTx() bool                          { return c.recheckTx }
-func (c Context) MinGasPrices() DecCoins                     { return c.minGasPrice }
-func (c Context) EventManager() EventManagerI                { return c.eventManager }
-func (c Context) Priority() int64                            { return c.priority }
-func (c Context) KVGasConfig() storetypes.GasConfig          { return c.kvGasConfig }
-func (c Context) TransientKVGasConfig() storetypes.GasConfig { return c.transientKVGasConfig }
+func (c Context) Context() context.Context                      { return c.baseCtx }
+func (c Context) MultiStore() storetypes.MultiStore             { return c.ms }
+func (c Context) BlockHeight() int64                            { return c.header.Height }
+func (c Context) BlockTime() time.Time                          { return c.header.Time }
+func (c Context) ChainID() string                               { return c.chainID }
+func (c Context) TxBytes() []byte                               { return c.txBytes }
+func (c Context) Logger() log.Logger                            { return c.logger }
+func (c Context) VoteInfos() []abci.VoteInfo                    { return c.voteInfo }
+func (c Context) GasMeter() storetypes.GasMeter                 { return c.gasMeter }
+func (c Context) BlockGasMeter() storetypes.GasMeter            { return c.blockGasMeter }
+func (c Context) IsCheckTx() bool                               { return c.checkTx }
+func (c Context) IsReCheckTx() bool                             { return c.recheckTx }
+func (c Context) MinGasPrices() DecCoins                        { return c.minGasPrice }
+func (c Context) EventManager() EventManagerI                   { return c.eventManager }
+func (c Context) Priority() int64                               { return c.priority }
+func (c Context) KVGasConfig() storetypes.GasConfig             { return c.kvGasConfig }
+func (c Context) TransientKVGasConfig() storetypes.GasConfig    { return c.transientKVGasConfig }
+func (c Context) StreamingManager() storetypes.StreamingManager { return c.streamingManager }
```

Comment:
if this is only being used by store should we pass it directly to the store instead of adding it to the context?

Reply:
Do you have an idea in mind? By having it here I was able to introduce the Context interface in the store (removing the sdk dependency) and having types/context implement the store Context interface to do a type conversion in the ABCIListener methods and retrieve the StreamingManager.

Reply:
lets leave it for now. In the future this will change, but no need to block this pr on that. Thank you